app.rb

```diff
@@ -0,0 +1,79 @@
+require_relative './lib/database_connection'
+require_relative './lib/item_repository'
+require_relative './lib/order_repository'
+require_relative './lib/item'
+require_relative './lib/order'
+
+class Application
+
+  def initialize(database_name, io, item_repository, order_repository)
+    DatabaseConnection.connect(database_name)
+    @io = io
+    @item_repository = item_repository
+    @order_repository = order_repository
+  end
+
+  def run
+    @io.puts "Welcome to the shop management program!"
+    @io.puts "\nWhat do you want to do?\n 1 = List all shop items\n 2 = Create a new item\n 3 = List all orders\n 4 = Create a new order\n"
+    number = @io.gets.chomp.to_i
+    fail "Please choose a valid option"  unless number == 1 || number == 2 || number == 3 || number == 4
```

Comment:
Catching user errors is very good. Perhaps loop the user back to the beginning as opposed to crashing the program?